resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -78,7 +78,8 @@ public static void writeNomapResponse(BuiltResponse jaxrsResponse, final HttpReq
       
       executeFilters(jaxrsResponse, request, response, providerFactory, method);
 
-      if (jaxrsResponse.getEntity() == null)
+      //[RESTEASY-1627] check on response.getOutputStream() to avoid resteasy-netty4 trying building a chunked response body for HEAD requests 
+      if (jaxrsResponse.getEntity() == null || response.getOutputStream() == null)

```

Comment:
In case of resteasy-netty, that's added later by RestEasyHttpResponseEncoder::encode, for resteasy-netty4 that's added later by NettyHttpResponse::getEmptyHttpResponse .
I don't know about wfly/eap and the other servers, but looks like we should fix them all in the relevant places (actually, I just figured out the resteasy-vertx is probably affected by this issue too, for example).